library/command/command/src/main/java/org/quiltmc/qsl/command/mixin/CommandManagerMixin.java

```diff
@@ -46,11 +61,42 @@
 			)
 	)
 	void addQuiltCommands(CommandManager.RegistrationEnvironment environment, CallbackInfo ci) {
-		CommandRegistrationCallback.EVENT.invoker().registerCommands(this.dispatcher, environmentMatches(environment, CommandManager.RegistrationEnvironment.INTEGRATED), environmentMatches(environment, CommandManager.RegistrationEnvironment.DEDICATED));
+		CommandRegistrationCallback.EVENT.invoker().registerCommands(this.dispatcher,
+				environmentMatches(environment, CommandManager.RegistrationEnvironment.INTEGRATED),
+				environmentMatches(environment, CommandManager.RegistrationEnvironment.DEDICATED)
+		);
 	}
 
 	@Unique
 	private static boolean environmentMatches(CommandManager.RegistrationEnvironment a, CommandManager.RegistrationEnvironment b) {
 		return a == b || b == CommandManager.RegistrationEnvironment.ALL;
 	}
+
+	// region Copyright 2016, 2017, 2018, 2019, 2020 zml and Colonel contributors.
+	// Licensed under the Apache 2.0 License; see this file's header for additional information.
+	@SuppressWarnings({"rawtypes", "unchecked"}) // argument type generics
+	@Inject(method = "makeTreeForSource", locals = LocalCapture.CAPTURE_FAILEXCEPTION,
```

Comment:
Why `FAILEXCEPTION` and not `FAILHARD`?

Reply:
No idea! Should I change it to `FAILHARD`?

Reply:
`FAILHARD` is used in other parts of QSL, it's probably best to be consistent (and to fail early)